pom.xml

```diff
@@ -1212,6 +1212,7 @@
     <properties>
         <maven.compiler.source>11</maven.compiler.source>
         <maven.compiler.target>11</maven.compiler.target>
+        <project.build.outputTimestamp>2025-04-09T17:14:21Z</project.build.outputTimestamp>
```

Comment:
thanks for the PR. does it mean the build time stamp is hardcoded? do we need to update it regularly somehow?

Reply:
it is updated by `versions:set` when changing from SNAPSHOT to release and vice versa, or by maven-release-plugin
or you can also choose not to update but pick a purely conventional value

Reply:
I recommend to set it instead to the timestamp of the latest git commit which can be done on every build
 <project.build.outputTimestamp>${git.commit.timestamp.datetime}</project.build.outputTimestamp>

using https://github.com/qoomon/maven-git-versioning-extension

Reply:
that's a classical reaction from many people, as it seems a legitimate idea

just know that by doing so, you are killing any cache, as every commit will change the hash of every output archive, even if nothing changed

changing timestamp only on version updates limits the times when the cache will be de-facto flushed for any output of a multi-module Maven build